ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -222,4 +222,50 @@ private BackupReplayFile newReplayFile() {
         ? Optional.of(new BackupReplayFile(latestBakFile))
         : Optional.empty();
   }
+
+  private void ensureDirectoryExists(final File backupsDir) {
+    if (!backupsDir.exists()) {
+      if (!backupsDir.mkdirs()) {
+        throw new KsqlServerException("Couldn't create the backups directory: "
+            + backupsDir.getPath()
+            + "\n Make sure the directory exists and is readable/writable for KSQL server "
+            + "\n or its parent directory is readable/writable by KSQL server"
+            + "\n or change it to a readable/writable directory by setting '"
+            + KsqlConfig.KSQL_METASTORE_BACKUP_LOCATION
+            + "' config in the properties file."
+        );
+      }
+
+      try {
+        Files.setPosixFilePermissions(backupsDir.toPath(),
+            PosixFilePermissions.fromString("rwx------"));
+      } catch (final IOException e) {
+        throw new KsqlServerException(String.format(
+            "Couldn't set POSIX permissions on the backups directory: %s. Error = %s",
+            backupsDir.getPath(), e.getMessage()));
+      }
+    }
+
+    if (!backupsDir.isDirectory()) {
+      throw new KsqlServerException(backupsDir.getPath()
+          + " is not a directory."
+          + "\n Make sure the directory exists and is readable/writable for KSQL server "
+          + "\n or its parent directory is readable/writable by KSQL server"
+          + "\n or change it to a readable/writable directory by setting '"
+          + KsqlConfig.KSQL_METASTORE_BACKUP_LOCATION
+          + "' config in the properties file."
+      );
+    }
+
+    if (!backupsDir.canWrite() || !backupsDir.canRead() || !backupsDir.canExecute()) {
```

Comment:
Why do we need execute permissions? I see we have a similar check for the Kafka Streams state directory so I assume there's good reason. Curious what it is.

Reply:
Directories need to have execute permissions in order to create files in it. By default, a new directory has `rwx`. If you remove the `x`, then you get a permissions denied.

```
$ mkdir /tmp/dir
$ chmod a-x /tmp/dir
$ ls -dl /tmp/dir
drw-rw-r-- 2 sergio sergio 4096 Jul 24 15:46 /tmp/dir
$ echo file > /tmp/dir/file
bash: /tmp/dir/file: Permission denied
$ 
```